### PR TITLE
#29, #32 Fix comment grammar, change comment tag to {{- -}}

### PR DESCRIPTION
**Pull request by kaygorodov**
- changed the comment grammar to properly tokenize comments.
- comments might be nested (if it is not needed, let me know I will change it back) like in Swift.
- added a few tests to check the cases that used to produce errors (text in between comments).
- changed the comment tag {{# to {{-. Do we need a backward compatibility here?

### CODE REVIEW

**Comment by kaygorodov** [comment]
@danpersa please review the changes.

File: src/test/scala/de/zalando/beard/renderer/BugSpec.scala [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
-
-import com.typesafe.scalalogging.LazyLogging
-import org.scalatest.{Matchers, FunSpec}
-
-import scala.io.Source
-
-/**
-  * @author dpersa
-  */
-class BugSpec extends FunSpec with Matchers with LazyLogging {
```
Comment (danpersa) [inline src/test/scala/de/zalando/beard/renderer/BugSpec.scala:11]: The goal of this test is to make sure that the bug never happens again... I guess you should rename it or move it to a better place, to better reflect what it tests, but you shouldn't remove it...

Comment (kaygorodov) [inline src/test/scala/de/zalando/beard/renderer/BugSpec.scala:11]: But I added the test that checks this case into the BeardTemplateParserSpec.scala. Check please:

https://github.com/kaygorodov/beard/blob/a059681bf23d5977b0e3b3ae8d352f5a9217d1fa/src/test/scala/de/zalando/beard/parser/BeardTemplateParserSpec.scala#L195

Comment (danpersa) [inline src/test/scala/de/zalando/beard/renderer/BugSpec.scala:11]: great :) didn't notice :)

**Pull request merged**